server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1514,6 +1514,13 @@ public void forceMerge(ForceMergeRequest forceMerge) throws IOException {
         engine.forceMerge(forceMerge.flush(), forceMerge.maxNumSegments(), forceMerge.onlyExpungeDeletes(), forceMerge.forceMergeUUID());
     }
 
+    public void triggerPendingMerges() throws IOException {
+        switch (state /* single volatile read */) {
+            case STARTED, POST_RECOVERY -> getEngine().forceMerge(false, ForceMergeRequest.Defaults.MAX_NUM_SEGMENTS, false, null);
```

Comment:
Nit: Can we add a comment about passing `flush=false` and `null` for the uuid? My understanding is that they do not make sense when the downstream code calls `IndexWriter#maybeMerge`. But it is not obvious from here.
 
Also for my knowledge: Since we don't flush actively, I guess it relies on flushes triggered by other mechanisms, such as scheduled refresh and indexing disk and memory controllers and maybe other things?

Reply:
More comments in c07131b08b0fd871ba814c8b805888a42bcc5150.

Henning and I are still discussing how to ensure we flush the result of the merge.